### PR TITLE
events: wire component name for each controller

### PR DESCRIPTION
**Pull request by mfojtik**
Builds on https://github.com/openshift/library-go/pull/301

This will add `-controller-name` suffix to event recorder component name for each controller we own.

### CODE REVIEW

File: pkg/operator/configobserver/config_observer_controller.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func NewConfigObserver(
 ) *ConfigObserver {
 	return &ConfigObserver{
 		operatorConfigClient: operatorClient,
-		eventRecorder:        eventRecorder,
+		eventRecorder:        eventRecorder.ForComponent(eventRecorder.ComponentName() + "-network-observer"),
```
Comment (sttts) [inline pkg/operator/configobserver/config_observer_controller.go]: is this network only?

Comment (mfojtik) [reply pkg/operator/configobserver/config_observer_controller.go]: good catch, fixed.

File: pkg/operator/unsupportedconfigoverridescontroller/unsupportedconfigoverrides_controller.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func NewUnsupportedConfigOverridesController(
 ) *UnsupportedConfigOverridesController {
 	c := &UnsupportedConfigOverridesController{
 		operatorClient: operatorClient,
-		eventRecorder:  eventRecorder,
+		eventRecorder:  eventRecorder.ForComponent(eventRecorder.ComponentName() + "-unsupportedconfigoverrides-controller"),
```
Comment (sttts) [inline pkg/operator/unsupportedconfigoverridescontroller/unsupportedconfigoverrides_controller.go]: with dashes like static-pod-state ?

**Pull request merged**